def foo():
    print(a)

foo()                         # error
a = 10 
